lib/BackgroundJobs/StorageCrawlJob.php

```diff
@@ -78,6 +93,28 @@ protected function run($argument): void {
 			$this->logger->error('Could not find storage root');
 			return;
 		}
+		
+
+		$directoryTypes = array_map(fn ($mimeType) => $this->mimeTypes->getId($mimeType), Constants::DIRECTORY_FORMATS);
+		$octetStreamTypes = array_map(fn ($mimeType) => $this->mimeTypes->getId($mimeType), Constants::OCTET_STREAM_FORMATS);
+
+		$qb = new CacheQueryBuilder($this->db, $this->systemConfig, $this->logger);
+		try {
+			$ignoreFiles = $qb->selectFileCache()
+				->andWhere($qb->expr()->in('mimetype', $qb->createNamedParameter($octetStreamTypes, IQueryBuilder::PARAM_INT_ARRAY)))
+				->andWhere($qb->expr()->in('name', $qb->createNamedParameter(['.nomedia', '.noimage'], IQueryBuilder::PARAM_STR_ARRAY)))
```

Comment:
Ideally, we should make sure that .noimage is only used for classifiers that consume IMAGE_FORMATS :/

Reply:
This would require extensive modification.
At least it was not implemented in v2.2.2.
https://github.com/nextcloud/recognize/blob/v2.2.2/lib/Service/ImagesFinderService.php#L17

Reply:
I know :/ It was implemented in v2, though: https://github.com/nextcloud/recognize/blob/v2.2.2/lib/Service/AudioFinderService.php#L17

Reply:
Oh, I didn't check it properly.
I implemented this, but using anything other than .nomedia to exclude large numbers of files may result in poor performance.